### PR TITLE
 syncer(dm): add collation config "collation_compatible"

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Issue Number: close  [#3420](https://github.com/pingcap/tiflow/issues/3420)

### What is changed and how it works?
add config `collation_compatible`.
default `loose ` handle create sql by original sql, will not add default collation as upstream.
`strick` will add default collation as upstream, and downstream will occur error when downstream don't support.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to update the documentation

### Release note 

```release-note
add config `collation_compatible`.
```

### CODE REVIEW

**Comment by WizardXiao** [comment]
i will fix integration test after add same config in dumpling.

File: dm/dm/config/subtask.go [hunk 1]
```diff
@@ -229,6 +229,10 @@ type SubTaskConfig struct {
 	// if case insensitive, we would convert schema/table name/pattern to lower case
 	CaseSensitive bool `toml:"case-sensitive" json:"case-sensitive"`
 
+	// default "loose" handle create sql by original sql, will not add default collation as upstream
+	// "strict" will add default collation as upstream, and downstream will occur error when downstream don't support
+	CollationCompatible string `yaml:"collation_compatible" toml:"collation_compatible" json:"collation_compatible"`
```
Comment (Ehco1996) [inline dm/dm/config/subtask.go:234]: could you please add a subtask in https://github.com/pingcap/tiflow/issues/3397  to remind us not to forget to add this new config to OpenAPI

**Review by Ehco1996 — COMMENTED** [review]
LGTM after fix the it test, do wee need add this config to doc ?

File: dm/dm/config/task.go [hunk 2]
```diff
@@ -357,6 +368,7 @@ func NewTaskConfig() *TaskConfig {
 		Loaders:                 make(map[string]*LoaderConfig),
 		Syncers:                 make(map[string]*SyncerConfig),
 		CleanDumpFile:           true,
+		CollationCompatible:     defaultCollationCompatible,
```
Comment (lance6716) [inline dm/dm/config/task.go:371]: taskConfig.adjust should check if given value is legal

File: dm/dm/config/task_test.go [hunk 3]
```diff
@@ -1025,7 +1025,7 @@ func (t *testConfig) TestTaskConfigForDowngrade(c *C) {
 	// make sure all new field were added
 	cfgReflect := reflect.Indirect(reflect.ValueOf(cfg))
 	cfgForDowngradeReflect := reflect.Indirect(reflect.ValueOf(cfgForDowngrade))
-	c.Assert(cfgReflect.NumField(), Equals, cfgForDowngradeReflect.NumField()+2) // without flag and tidb
+	c.Assert(cfgReflect.NumField(), Equals, cfgForDowngradeReflect.NumField()+3) // without flag and tidb
```
Comment (lance6716) [inline dm/dm/config/task_test.go:1028]:
```suggestion
	c.Assert(cfgReflect.NumField(), Equals, cfgForDowngradeReflect.NumField()+3) // without flag, tidb and collation_compatible
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: dm/dm/config/task.go (hunk 2 above)
Comment (WizardXiao) [reply dm/dm/config/task.go:371]: ok

File: dm/dm/config/task_test.go (hunk 3 above)
Comment (WizardXiao) [reply dm/dm/config/task_test.go:1028]: ok

File: dm/dm/config/subtask.go (hunk 1 above)
Comment (WizardXiao) [reply dm/dm/config/subtask.go:234]: ok

**Comment by WizardXiao** [comment]
> LGTM after fix the it test, do wee need add this config to doc ?

yes, this need add to doc, i will write doc pr.

**Review by lance6716 — APPROVED** [review]

**Review by Ehco1996 — APPROVED** [review]

**Pull request merged**